staging/src/k8s.io/component-helpers/scheduling/corev1/nodeaffinity/nodeaffinity.go

```diff
@@ -287,3 +287,42 @@ type preferredSchedulingTerm struct {
 	nodeSelectorTerm
 	weight int
 }
+
+type RequiredSchedulingTerm struct {
+	requiredNodeSelector labels.Selector
+	requiredNodeAffinity *LazyErrorNodeSelector
+}
+
+// GetPodRequiredNodeSelectorAndAffinity returns the parsing result of pod's nodeSelector and nodeAffinity.
+func GetPodRequiredNodeSelectorAndAffinity(pod *v1.Pod) *RequiredSchedulingTerm {
+	var selector labels.Selector
+	if len(pod.Spec.NodeSelector) > 0 {
+		selector = labels.SelectorFromSet(pod.Spec.NodeSelector)
+	}
+	// Use LazyErrorNodeSelector for backwards compatibility of parsing errors.
+	var affinity *LazyErrorNodeSelector
+	if pod.Spec.Affinity != nil &&
+		pod.Spec.Affinity.NodeAffinity != nil &&
+		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+		affinity = NewLazyErrorNodeSelector(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution)
+	}
+	return &RequiredSchedulingTerm{requiredNodeSelector: selector, requiredNodeAffinity: affinity}
+}
+
+// PodMatchesNodeSelectorAndAffinityTerms checks whether the pod is schedulable onto nodes according to
+// the requirements in both nodeSelector and nodeAffinity.
+func PodMatchesNodeSelectorAndAffinityTerms(s *RequiredSchedulingTerm, node *v1.Node) bool {
```

Comment:
This should be a method of `RequiredSchedulingTerm` (or `NodeAffinity` as suggested above), just called `Match`, or even `NoErrorMatch`. Document in the comment above what "NoError" refers to. I basically want to avoid new users of this function.